### PR TITLE
Single Line Text Field

### LINKED ISSUE #145: [UI Component] Custom Single line textfield
**Requirements:**
- The textfield displays only a single line
- in the case of text has a middle  truncation mode
- allows to be titled
- the title can have an accessory view that performs an action
- the textfield allows one or more trailing accessory views
- the textfield can be provided with a validation function
- the textfield can display a caption/footer with relevant information about the current or future input
- the textfield can display a placeholder value that clears out when user input is detected

**Scoping questions**
Shall the title and its accessory view be left out of this component?

**Samples**

### PR DESCRIPTION
**Pull request by adamstener**
Closes #145 
CleanShot 2022-04-06 at 07 10 45

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [x] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [x] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secant/Util/Strings.swift [hunk 1]
```diff
@@ -8,3 +8,56 @@ extension String {
         self.init(stringLiteral: output)
     }
 }
+
+extension String {
+    private static let floatingPointRegex = "^[0-9]*.?[0-9]+"
```
Comment (pacu) [inline secant/Util/Strings.swift:13]: This regex might be an example, but it's wrong. In many languages the dot is the group separator, not the decimal separator.

**Review by pacu — CHANGES_REQUESTED** [review]
Great work @adamstener 

Left a couple of comments and questions.

File: secant/Colors.xcassets/Cursor/Bar.colorset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "colors" : [
+    {
+      "color" : {
+        "color-space" : "display-p3",
```
[the remark this answers is not in the document]
Comment (adamstener) [reply secant/Colors.xcassets/Cursor/Bar.colorset/Contents.json:5]: This is generated code

File: secant/Util/Strings.swift (hunk 1 above)
Comment (adamstener) [reply secant/Util/Strings.swift:13]: I just put a placeholder in here, it would obviously be trivial to update these regexes as you need

Comment (pacu) [reply secant/Util/Strings.swift:13]: sounds fair. Let's add an issue referencing that an a todo then so we don't miss it 👍

**Review by pacu — APPROVED** [review]
Looks good to me!


Just remains adding a TODO to the placeholder regex with the corresponding issue.

File: secant/Util/Strings.swift (hunk 1 above)
Comment (pacu) [inline secant/Util/Strings.swift:13]:
```suggestion
    // TODO: Issue #245 Add Validation Regex that support localization 
    private static let floatingPointRegex = "^[0-9]*.?[0-9]+"
```

**Pull request merged**